### PR TITLE
UI 틀을 다시 잡았습니다.

### PR DESCRIPTION
**Pull request by chepchep2**
많이 헤매고 있는데 일단 UI틀을 다시 잡았습니다.
구현 결과입니다. 
- https://imgur.com/vwvuSMW

### CODE REVIEW

**Comment by chepchep2** [comment]
* **#92** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/92?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#91** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/91?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#90** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/90?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#89** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/89?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#88** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/88?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#87** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/87?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#86** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/86?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#85** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/85?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#84** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/84?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#83** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/83?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#82** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/82?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#81** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/81?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#80** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/80?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#79** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/79?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#78** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/78?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#77** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/77?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#75** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/75?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#74** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/74?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#73** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/73?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#72** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/72?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#71** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/71?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#70** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/70?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#69** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/69?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/mainPage`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @chepchep2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/common/game_card.dart [hunk 1]
```diff
@@ -44,7 +47,7 @@ class GameCard extends StatelessWidget {
     this.btnTitleStyle,
   });
 
-  String getMatchStatus(BuildContext context) {
+  Widget getMatchStatus(BuildContext context) {
```
Comment (flowkater) [inline lib/common/game_card.dart:50]: 위젯 클래스 바깥으로 따로 분리해서 StatelessWidget 으로 만드는게 좋아보입니다.

아래 if 문은 

```dart
final now = DateTime.now();
  final today = DateTime(now.year, now.month, now.day);
  final tomorrow = today.add(const Duration(days: 1));
  final diff = matchDate.difference(today).inDays;

  final matchStartDateTime = DateTime(matchDate.year, matchDate.month, matchDate.day, startTime.hour, startTime.minute);
  final matchEndDateTime = DateTime(matchDate.year, matchDate.month, matchDate.day, endTime.hour, endTime.minute);

  if (matchDate.isBefore(today)) {
    return '';
  }

  if (matchDate == today && now.isAfter(matchStartDateTime) && now.isBefore(matchEndDateTime)) {
    return '진행중';
  }

  final formattedDate = DateFormat('M월 d일 E', 'ko_KR').format(matchDate);
  final formattedTime = _formatTime(startTime);

 if (diff == 0) {
  return '${_formatTime(startTime)} 시작';
} else if (diff == 1) {
  return '내일 $formattedDate • $formattedTime';
} else if (diff <= 100) {
  return 'D-$diff $formattedDate • $formattedTime';
} else {
  return '$formattedDate • $formattedTime';
}
```


위 코드 참고해서 리팩토링 하는게 좋아보입니다.
다만, 위젯을 반환해야한다면,
최종 상태값을 조건에 따라 반환하고 그 상태값에 맞는 위젯을 매핑해서 반환하는게 좋아보여요.


getMatchStatus -> enum 상태값만 return
enum 상태 값에 따라서 적절한 위젯을 _private function 으로 잘 쪼개서 반환

File: lib/common/game_card.dart [hunk 2]
```diff
@@ -81,49 +202,189 @@ class GameCard extends StatelessWidget {
     return '$amPm $formattedHour시 $formattedMinute분';
   }
 
+  List<Widget> getPositionChips(List<String> positions) {
```
Comment (flowkater) [inline lib/common/game_card.dart:205]: 이것도 위처럼 분리하는게 나아보입니다.
리팩토링은 제가 적은 코멘트 참고해보세요

File: lib/page/main_page.dart [hunk 3]
```diff
@@ -21,108 +21,166 @@ class _MainPageState extends State<MainPage>
   final PageController _pageController = PageController();
   late TabController _tabController;
 
+  int _selectedIndex = 0;
+  DateTime _currentTime = DateTime.now();
+  late Timer _timer;
+
   final List<GameCard> _upcomingGames = [
```
Comment (flowkater) [inline lib/page/main_page.dart:28]: 별로 파일로 빼면 좋을 것 같습니다. repository 등

File: lib/page/main_page.dart [hunk 4]
```diff
@@ -342,60 +406,100 @@ class _MainPageState extends State<MainPage>
...
+                                  const Text('다음 경기 일정',
+                                      style: TextStyle(
+                                          fontSize: 18,
+                                          fontWeight: FontWeight.bold)),
+                                  Text('${filteredUpcomingGames.length} >',
+                                      style: const TextStyle(fontSize: 14)),
+                                ],
+                              ),
+                            ),
+                            ...filteredUpcomingGames.map((game) {
```
Comment (flowkater) [inline lib/page/main_page.dart:476]: ...todayGames 랑 ...filteredUpcomingGames 랑 만 다르고, 코드가 겹치는 것 같습니다.
private 위젯 함수로 분리하면 좋을 것 같습니다.

File: lib/page/main_page.dart [hunk 5]
```diff
@@ -184,7 +242,14 @@ class _MainPageState extends State<MainPage>
 
   @override
   Widget build(BuildContext context) {
-    final todayGames = _getTodayGames(_upcomingGames);
+    final today = DateTime.now();
+    final sortedUpcomingGames = List<GameCard>.from(_upcomingGames)
+      ..sort((a, b) => a.matchDate.compareTo(b.matchDate));
+
+    final todayGames = _getTodayGames(sortedUpcomingGames);
+    final filteredUpcomingGames = sortedUpcomingGames
```
Comment (flowkater) [inline lib/page/main_page.dart:250]: 해당 값들은 전부 initState 에서 state 로 추후에는 셋업을 해주면 좋을 것 같습니다.
아직 UI 작업 중이라 그렇겠지만, 여기에 변수를 선언하면 UI가 업데이트될때마다 매번 새로 변수가 셋업이 됩니다.

**Review by flowkater — CHANGES_REQUESTED** [review]
일단 중간 과정을 거치는 코드들이 많아서 최종 브랜치에만 코드 리뷰 드립니다.
필요한 부분은 제가 코드를 받아서 다시 한번 찬찬히 살펴보겠습니다.

**Comment by flowkater** [comment]
### Merge activity

* **May 15, 7:11 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/75) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**